src/KubernetesClient/Authentication/GcpTokenProvider.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Diagnostics;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using k8s.Exceptions;
+using Microsoft.Rest;
+using Newtonsoft.Json.Linq;
+
+namespace k8s.Authentication
+{
+    public class GcpTokenProvider : ITokenProvider
```

Comment:
This file needs a unit test.

Reply:
How would you expect the unit test to look? This class is 90% call to an external process which a) doesn't exist in unit test environment
b) requires to be authenticated / configured in order to respond correctly

I can certainly try to create a shell script to mock of what gcloud cli returns, though I question the value of such test. Beyond invoking external command, the rest of this class is just two lines of json querying.

Reply:
There's sufficient logic (e.g. return code != 0 -> failure, etc) that I think it deserves a test.

There's an equivalent test for the Javascript client here:

https://github.com/kubernetes-client/javascript/blob/master/src/exec_auth_test.ts